src/ui/QuerySummary/QuerySummary.ts

```diff
@@ -189,19 +241,101 @@ export class QuerySummary extends Component {
     }
   }
 
+  private parseNoResultsFoundMessage(noResultsFoundMessage: string) {
+    if (this.isQuerySummaryTagInMessage(noResultsFoundMessage)) {
+      let messageSections = noResultsFoundMessage.split(QUERY_TAG);
```

Comment:
Since we would move it to use `${query}` instead of `<% %>` and it would match what we have in `ResultLink`, I would extract the method used in `ResultLink` in a separate module or utility function, and use the same logic for both components.

All this logic you have here could then go away, and be replaced by a cleaner `regex` replace.

Reply:
Good idea! 
I checked in the ResuktsLink.ts file the method to extract which is : parseStringTemplate().
There is one little problem, the method execute the same logic, but for a string. In the scenario of paseNoResultsFoundMessage() it's an array of Dom elements because we want to apply the coveo-highlight css class to the ${query} text.  I see two options : 
 - We could extract the method from ResultsLink.ts, change the logic of it by adding a boolean who would specify if we want the value in the ${ } tags to be bold or not. Apply the bold text with regex.
 - We could keep the method in ResultsLink.ts and just copy the majority of the logic of it in QuerySummary.ts to prevent the parseStringTemplate() method from becoming more complex just for a specific use case.

Or certainly, maybe another option?